crates/ruff_python_ast/src/source_code/indexer.rs

```diff
@@ -18,17 +21,32 @@ impl Indexer {
     pub fn continuation_lines(&self) -> &[usize] {
         &self.continuation_lines
     }
+
+    pub fn string_lines(&self) -> &[Range] {
+        &self.string_lines
+    }
 }
 
 impl From<&[LexResult]> for Indexer {
     fn from(lxr: &[LexResult]) -> Self {
         let mut commented_lines = Vec::new();
         let mut continuation_lines = Vec::new();
+        let mut string_lines = Vec::new();
         let mut prev: Option<(&Location, &Tok, &Location)> = None;
         for (start, tok, end) in lxr.iter().flatten() {
             if matches!(tok, Tok::Comment(_)) {
                 commented_lines.push(start.row());
             }
+            if matches!(
+                tok,
+                Tok::String {
+                    value: _,
+                    kind: _,
+                    triple_quoted: true
+                }
+            ) {
+                string_lines.push(Range::new(*start, *end));
+            }
```

Comment:
Nit:
```suggestion
			match tok {
				Tok::Comment(_) => {
	                commented_lines.push(start.row());
	            }
                Tok::String {
                    value: _,
                    kind: _,
                    triple_quoted: true
                } => {
                	string_lines.push(Range::new(*start, *end));
	            }
```

To express that the branches are mutually exclusive